backend/src/auth/passport.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  NextFunction,
+  Request,
+  Response
+} from "express";
+import _ from "lodash";
+import passport from "passport";
+import passportLocal from "passport-local";
+import {
+  getConnection,
+  Repository
+} from 'typeorm';
+import {
+  default as User
+} from "../entity/user.entity";
+
+const localStrategy = passportLocal.Strategy;
+
+// Create a passport middleware to handle user registration
+passport.use('signup', new localStrategy({
+  usernameField: 'email',
+  passwordField: 'password'
+}, async (email: string, password: string, done: any) => {
+  try {
+    const user = new User();
+    user.email = email;
+    user.password = password;
+
+    await user.save();
+
+    // Send the user information to the next middleware
+    return done(null, user);
+  } catch (error) {
+    done(error);
+  }
+}));
+
+// Create a passport middleware to handle User login
+passport.use('login', new localStrategy({
+  usernameField: 'email',
+  passwordField: 'password'
+}, async (email: string, password: string, done: any) => {
+  try {
+    // Find the user associated with the email provided by the user
+    const user = await User.findOne({
+      email
+    });
+    if (!user) {
+      // If the user isn't found in the database, return a message
+      return done(null, false, {
+        message: 'User not found'
```

Comment:
We should probably change this later so that it doesn't tell anyone if the account already exists. It should just say 'Either user or password is incorrect'.

But we can change that later